python/hail/representation/interval.py

```diff
@@ -113,7 +127,16 @@ def end(self):
 
         :rtype: :class:`.Locus`
         """
-        return Locus._from_java(self._jrep.end())
+        return Locus._from_java(self._jrep.end(), self._rg)
+
+    @property
+    @record_method
+    def reference_genome(self):
```

Comment:
I feel like there's some weirdness here where if you call .reference_genome a bunch of times it just adds on to history of the existing reference genome, even though it's never modified. e.g. (bad example, and maybe the answer is just that this kind of thing shouldn't ever be done)
```
r = Locus.parse('1:101230').reference_genome
l3 = Locus.parse('1:101230', reference_genome = r).reference_genome
```
will print out a bunch of stuff, even though it's just the default reference. Does it not work with the default reference stuff to just have the history be created upon GenomeReference creation?

Reply:
I think this behavior is fine. If the user has a pipeline that is correct, then we're just capturing exactly what they're running.

Reply:
ok, sounds good!